Tests/ML/test_download_upload.py

```diff
@@ -212,9 +224,31 @@ def test_mount_in_azureml2(test_output_dirs: OutputFolderForTests) -> None:
 def test_mount_or_download(test_output_dirs: OutputFolderForTests) -> None:
     """
     Tests the different combinations of local and Azure datasets, with Innereye built-in and container models.
+    First files from 'data.csv' are copied to the mounted, downloaded and root folders.
     """
     root = test_output_dirs.root_dir
+
+    # With runs outside of AzureML, a local dataset should be used as-is. Azure dataset ID is ignored here.
+    shutil.copy(full_ml_test_data_path(DATASET_CSV_FILE_NAME), root / DATASET_CSV_FILE_NAME)
+
+    df = pandas.read_csv(str(full_ml_test_data_path(DATASET_CSV_FILE_NAME)), usecols=['filePath'])
+
     for is_lightning_model in [True, False]:
+        if not is_lightning_model:
+            # Consider three directories, i) "downloaded", ii)  "mounted", iii) "train_and_test_data" Path for ""
+            # represents "train_and_test_data" and is empty since the string "train_and_test_data" is in data.csv
+            # file and corresponds to the case container.local_dataset == root
+            for myPath in ["downloaded", "mounted", ""]:
+
+                path = test_output_dirs.root_dir / myPath
+                path.mkdir(exist_ok=True)
+
+                train_and_test_data_path = path / "train_and_test_data"
+                train_and_test_data_path.mkdir(exist_ok=True)
+
+                for filePath in set(df['filePath'].values):
+                    shutil.copy(full_ml_test_data_path() / filePath, path / filePath)
```

Comment:
as discussed, it would be simpler to make a change in _test_mount_for_lightning_container. There, the dataset.csv file is already copied to both "mounted" and "downloaded" folders.
```shutil.copytree(full_ml_test_data_path("train_and_test_data"), path)``` or ```shutil.copytree(full_ml_test_data_path("train_and_test_data"), path / "train_and_test_data")```

Reply:
Implemented changes to recursively copy full_ml_test_data_path("train_and_test_data") to path / "train_and_test_data"